### PR TITLE
(MODULES-3704) Update gemfile template like windows

### PR DESCRIPTION
**Pull request by hunner**

### CODE REVIEW

**Comment by hunner** [comment]
This is a subset of the work done in #60

**Comment by hunner** [comment]
Still WIP, too.

**Comment by glennsarti** [comment]
Apart from the rebase...Looks like a good start!

**Comment by ferventcoder** [comment]
Looks like a good start!

File: moduleroot/Gemfile [hunk 1]
```diff
@@ -2,57 +2,172 @@
...
+  elsif place_or_version =~ /^file:/
+    :file
   else
-    default_location
+    :gem
   end
 end
 
-<%
-def build_location(gem)
```
Comment (ferventcoder) [inline moduleroot/Gemfile]: Is this still needed or was that changed based on the ruby-operator, ruby-version removals?

File: config_defaults.yml [hunk 2]
```diff
@@ -46,29 +46,23 @@ Gemfile:
       - gem: parallel_tests
       - gem: rubocop
         version: '0.41.2'
-        ruby-operator: '<'
-        ruby-version: '2.0.0'
+        condition: "RUBY_VERSION < '2.0.0'"
```
Comment (DavidS) [inline config_defaults.yml]: I know we don't do it yet, but using `Gem::Version` would be great for correct semver comparisons.

File: moduleroot/Gemfile (hunk 1 above)
Comment (ferventcoder) [inline moduleroot/Gemfile]: I like that this helped build out the gem information versus continually creating a longer string value.

Comment (ferventcoder) [inline moduleroot/Gemfile]: But I see you cleaned it up some below from our template and yours, +1!

File: moduleroot/Gemfile [hunk 3]
```diff
@@ -2,57 +2,172 @@
...
+  else -%><%= ", '#{gem['version']}'" if gem['version'] %><%
+  end %>, <%=
+  ' ' * (maxlen - gem['length'])
+  %>:require => false<%=
+  ", :git => '#{gem['git']}'" if gem['git'] %><%=
+  ", :branch => '#{gem['branch']}'" if gem['branch'] %><%=
+  ", :ref => '#{gem['ref']}'" if gem['ref'] %><%=
+  ", :platforms => #{gem['platforms']}" if gem['platforms'] %><%=
+  " if #{gem['condition']}" if gem['condition'] %>
+<% end -%>
```
Comment (ferventcoder) [inline moduleroot/Gemfile]: I like the way that this got cleaned up!

File: config_defaults.yml (hunk 2 above)
Comment (ferventcoder) [inline config_defaults.yml]: That would be great!

**Comment by ferventcoder** [comment]
LGTM. Rekicked Travis build.

**Comment by ferventcoder** [comment]
@hunner probably unrelated but travis is failing on


~~~sh
bundler: failed to load command: msync (/home/travis/build/puppetlabs/modulesync_configs/vendor/bundle/ruby/2.3.0/bin/msync)
~~~

**Comment by hunner** [comment]
https://github.com/puppetlabs/puppetlabs-accounts/pull/71

File: config_defaults.yml [hunk 4]
```diff
@@ -29,13 +29,25 @@
...
+  supports_windows:
+    - puppetlabs-acl
+    - puppetlabs-chocolatey
+    - puppetlabs-dism
+    - puppetlabs-dsc
+    - puppetlabs-powershell
+    - puppetlabs-reboot
+    - puppetlabs-registry
+    - puppetlabs-sqlserver
+    - puppetlabs-wsus_client
```
Comment (ferventcoder) [inline config_defaults.yml]: That's an interesting choice. I would have assumed this was going to be in a different file.

Comment (DavidS) [inline config_defaults.yml]: I would prefer `supports_windows` to go away completely, but this is also in scope for MODULES-4027 (create a set of gems encapsulating our ruby-version-specific dependencies), so I'm supportive of adding a ugly hack here to get this merged.

Comment (hunner) [inline config_defaults.yml]: @ferventcoder `@configs` does not expose variables outside of the `config_defaults.yml` structure under the `Gemfile` key to the template as far as I saw, so this was the only location without making the template do `File.read`s, which would be worse...

Comment (hunner) [inline config_defaults.yml]: @glennsarti Both @ferventcoder and I had a discussion in hipchat yesterday about how `.sync.yml` should be centralized and go away entirely, but that is a project for another time.

In the mean time, it is modulesync_configs that cares about which modules receive Gemfile template variants, and the modules themselves should be ignorant of "fiddly bits" inside of the templates anyway, so the quality of extra template contents should live with the templates themselves.

Comment (hunner) [inline config_defaults.yml]: @DavidS Yes! It should go away entirely; see @ferventcoder's and my comment as such at https://github.com/puppetlabs/modulesync_configs/pull/99#discussion_r86820942

**Pull request merged**